lib/ex_unit/lib/ex_unit.ex

```diff
@@ -360,6 +360,30 @@ defmodule ExUnit do
     configure(after_suite: [function | current_callbacks])
   end
 
+  @doc """
+  Fetches the test supervisor for the current test.
+
+  This is the same supervisor as used by `ExUnit.Callbacks.start_supervised/2`
+  and similar, see `ExUnit.Callbacks` module documentation for more information.
+  """
+  @doc since: "1.11.0"
+  @spec fetch_test_supervisor!() :: pid()
+  def fetch_test_supervisor!() do
```

Comment:
I think this should go in the `ExUnit.Callbacks` module. I mean, in the same place where `start_supervised` here, rather :P

Reply:
we were working on this with José, we think this will be used rather infrequently so the idea to put it here was to not pollute user tests cases.

Reply:
Agreed it's not going to be used often. Still, I think semantically it belongs in `ExUnit.Callbacks`. Not a strong preference so up to you folks :)

Reply:
@whatyouhide I am fine with both too. :P @fertapric @ericmj does any of you have a pref?